File: src/main/java/logic/ItemBullet.java
package logic;

import lib.DrawingUtility;

public class ItemBullet /* fill code */{

	public ItemBullet(int movingDuration, int z) {
		/* fill code */	
	}

	@Override
	public void render() {
		DrawingUtility.drawItemBullet(x, y, radius, isPointerOver);
	}

	@Override
	public void collect(PlayerStatus player) {
		/* fill code */
	}

}

File: src/logic/ItemBullet.kt
package logic

import lib.DrawingUtility

class ItemBullet(movingDuration: Int, z: Int) : CollectibleObject(50, movingDuration, z, 30) {
    override fun render() = DrawingUtility.drawItemBullet(x, y, radius, isPointerOver)

    override fun collect(player: PlayerStatus) {
        if (player.currentGun is SpecialGun)
            (player.currentGun as SpecialGun).bulletQuantity = Int.MAX_VALUE
    }
}
